drivers/d3d12/rendering_device_d3d12.cpp

```diff
@@ -0,0 +1,10115 @@
+
+	read_offset += sizeof(uint32_t) * 3 + bin_data_size;
+
+	String name;
+
+	if (binary_data.shader_name_len) {
+		name.parse_utf8((const char *)(binptr + read_offset), binary_data.shader_name_len);
+		read_offset += binary_data.shader_name_len;
+		if ((binary_data.shader_name_len % 4) != 0) { // Alignment rules are really strange.
+			read_offset += 4 - (binary_data.shader_name_len % 4);
+		}
+	}
+
+	Vector<Shader::Set> set_info;
+	set_info.resize(binary_data.set_count);
+
+	for (uint32_t i = 0; i < binary_data.set_count; i++) {
+		ERR_FAIL_COND_V(read_offset + sizeof(uint32_t) >= binsize, RID());
+		uint32_t set_count = decode_uint32(binptr + read_offset);
+		read_offset += sizeof(uint32_t);
+		const RenderingDeviceD3D12ShaderBinaryDataBinding *set_ptr = reinterpret_cast<const RenderingDeviceD3D12ShaderBinaryDataBinding *>(binptr + read_offset);
+		uint32_t set_size = set_count * sizeof(RenderingDeviceD3D12ShaderBinaryDataBinding);
+		ERR_FAIL_COND_V(read_offset + set_size >= binsize, RID());
+
+		for (uint32_t j = 0; j < set_count; j++) {
+			Shader::ShaderUniformInfo sui;
+
+			sui.info.type = UniformType(set_ptr[j].type);
+			sui.info.writable = set_ptr[j].writable;
+			sui.info.length = set_ptr[j].length;
+			sui.info.binding = set_ptr[j].binding;
+
+			sui.binding.stages = set_ptr[j].stages;
+			static_assert(sizeof(UniformBindingInfo::root_sig_locations) == sizeof(RenderingDeviceD3D12ShaderBinaryDataBinding::root_sig_locations));
+			memcpy(&sui.binding.root_sig_locations, &set_ptr[j].root_sig_locations, sizeof(UniformBindingInfo::root_sig_locations));
+
+			set_info.write[i].uniforms.push_back(sui);
+		}
+
+		read_offset += set_size;
+	}
+
+	ERR_FAIL_COND_V(read_offset + binary_data.specialization_constant_count * sizeof(RenderingDeviceD3D12ShaderBinarySpecializationConstant) >= binsize, RID());
+
+	Vector<Shader::SpecializationConstant> specialization_constants;
+
+	for (uint32_t i = 0; i < binary_data.specialization_constant_count; i++) {
+		const RenderingDeviceD3D12ShaderBinarySpecializationConstant &src_sc = *(reinterpret_cast<const RenderingDeviceD3D12ShaderBinarySpecializationConstant *>(binptr + read_offset));
+		Shader::SpecializationConstant sc;
+		sc.constant.int_value = src_sc.int_value;
+		sc.constant.type = PipelineSpecializationConstantType(src_sc.type);
+		sc.constant.constant_id = src_sc.constant_id;
+		memcpy(sc.stages_bit_offsets, src_sc.stages_bit_offsets, sizeof(sc.stages_bit_offsets));
+		specialization_constants.push_back(sc);
+
+		read_offset += sizeof(RenderingDeviceD3D12ShaderBinarySpecializationConstant);
+	}
+
+	HashMap<ShaderStage, Vector<uint8_t>> stages_bytecode;
+
+	for (uint32_t i = 0; i < binary_data.stage_count; i++) {
+		ERR_FAIL_COND_V(read_offset + sizeof(uint32_t) * 3 >= binsize, RID());
+		uint32_t stage = decode_uint32(binptr + read_offset);
+		read_offset += sizeof(uint32_t);
+		uint32_t dxil_size = decode_uint32(binp
```

Comment:
```suggestion
					rtv_clears[num_rtv_clears++] = RTVClear{ handle, clear_color };
```

This is taking a pointer to a temporary which is undefined once the scope is left. This was breaking viewport clear colors for me when MSAA is enabled.